server/main.go

```diff
@@ -1,98 +1,145 @@
 package main
 
 import (
-	"encoding/json"
-	"fmt"
-	"io/ioutil"
-	"log"
-	"net/http"
-
-	"github.com/gorilla/mux"
+  "log"
+  "github.com/gin-gonic/gin"
+  "github.com/jinzhu/gorm"
+  _ "github.com/jinzhu/gorm/dialects/postgres"
 )
 
-type event struct {
-	ID          string `json:"ID"`
-	Title       string `json:"Title"`
-	Description string `json:"Description"`
+type User struct {
+  Id        int    `gorm:"AUTO_INCREMENT" form:"id" json:"id"`
+  FirstName string `gorm:"not null" form:"first_name" json:"first_name"`
+  LastName  string `gorm:"not null" form:"last_name" json:"last_name"`
 }
 
-type allEvents []event
+// Connect to DB
+func initDB() *gorm.DB {
+  // Openning file
+  db, err := gorm.Open("postgres", "host=postgres port=5432 dbname=public user=root sslmode=disable")
+  db.LogMode(true)
+  // Error
+  if err != nil {
+      panic(err)
+  }
+  // Creating the table
+  if !db.HasTable(&User{}) {
+      db.CreateTable(&User{})
+      db.Set("gorm:table_options", "ENGINE=InnoDB").CreateTable(&User{})
+  }
 
-var events = allEvents{
-	{
-		ID:          "1",
-		Title:       "Hello Shutterfly!",
-		Description: "Ipsum reprehenderit consectetur do commodo ea cupidatat magna non anim culpa culpa irure occaecat.",
-	},
+  return db
 }
 
-func createEvent(w http.ResponseWriter, r *http.Request) {
-	var newEvent event
-	reqBody, err := ioutil.ReadAll(r.Body)
+// Log errors
+func checkErr(err error, msg string) {
 	if err != nil {
-		fmt.Fprintf(w, "Please make sure the title and description are filled out")
+		log.Fatalln(msg, err)
 	}
-	json.Unmarshal(reqBody, &newEvent)
-	events = append(events, newEvent)
-	w.WriteHeader(http.StatusCreated)
-	json.NewEncoder(w).Encode(newEvent)
 }
 
-func updateEvent(w http.ResponseWriter, r *http.Request) {
-	eventID := mux.Vars(r)["id"]
-	var updatedEvent event
-	reqBody, err := ioutil.ReadAll(r.Body)
+// Group the routes
+func main() {
+  // initDB()
+  routes := gin.Default()
+  v1 := routes.Group("api/v1")
+  {
+    v1.POST("/users", PostUser)
+    v1.GET("/users", GetUsers)
+    v1.GET("/users/:id", GetUser)
+    v1.PUT("/users/:id", UpdateUser)
+    v1.DELETE("/users/:id", DeleteUser)
+  }
+  routes.Run(":8080")
+}
 
-	if err != nil {
-		fmt.Fprintf(w, "Please make sure the title and description are filled out")
-	}
+func PostUser(ctx *gin.Context) {
+    db := initDB()
```

Comment:
Why are we initializing the database on every request?

Reply:
I was getting errors for some reason, but this may get fixed when calling from the main function